llvm/test/Transforms/LoopUnroll/AArch64/scalable-vec-ins-ext.ll

```diff
@@ -0,0 +1,87 @@
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --version 3
+; RUN: opt -passes=loop-unroll,simplifycfg -S -mtriple aarch64 -mattr=+sve -aarch64-sve-vector-bits-min=128 %s | FileCheck %s -check-prefix=UNROLL-128
```

Comment:
I'm not sure about this runline. I used it to make sure `<8 x float>` was not a legal type, but if 128b is the actual size of the vectors at runtime then we won't really be able to insert that much data. Should this be ignored, or be rejected by the verifier if we know that it may be a invalid operation?

I think I can write another test for a negative case, e.g. one with non-power-of-2 integers that would require promotion rather than being directly legal.

This work might need to be extended in future to cover cases where we would legalize both sides in the same way, but it's a start for now.

Reply:
Could you test the cost of the intrinsics more directly, like the tests in llvm/test/Analysis/CostModel/AArch64/sve-intrinsics.ll or llvm/test/Analysis/CostModel/AArch64/splice.ll?